packages/starlight/integrations/virtual-user-config.ts

```diff
@@ -10,15 +10,15 @@ function resolveVirtualModuleId<T extends string>(id: T): `\0${T}` {
 /** Vite plugin that exposes Starlight user config and project context via virtual modules. */
 export function vitePluginStarlightUserConfig(
 	opts: StarlightConfig,
-	{ root, srcDir }: Pick<AstroConfig, 'root' | 'srcDir'>
+	{ root, srcDir, build }: Pick<AstroConfig, 'root' | 'srcDir' | 'build' >
```

Comment:
Ah yeah, good point about the assertions. They’re not the end of the world, but maybe we can type more accurately what we actually need here, so we can avoid them:

```suggestion
	{ root, srcDir, build }: Pick<AstroConfig, 'root' | 'srcDir'> & { build: Pick<AstroConfig['build'], 'format'> }
```